src/main/java/com/aws/greengrass/clientdevices/auth/session/SessionManager.java

```diff
@@ -63,6 +64,7 @@ public Session findSession(String sessionId) {
      * @return session id
      * @throws AuthenticationException if device credentials were not able to be validated
      */
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
```

Comment:
Is there a reason you added this here instead of in the API class with the other service errors? I don't have a super strong preference, but it may make things easier to find if we co-locate

Reply:
It was the first one I did so I ended up going a layer deeper, but I agree it'd be better to have it in the API class. I'll make that change